ykrt/src/compile/jitc_yk/aot_ir.rs

```diff
@@ -74,12 +78,26 @@ macro_rules! index {
 pub(crate) struct FuncIdx(usize);
 index!(FuncIdx);
 
+impl FuncIdx {
```

Comment:
Putting these functions on the *indexes* feels really weird (we got rid of this in a recent PR). How difficult would it be to put these functions on `Module`, which is where they belong?

Reply:
OK, so fa3c3cd3 inverts this logic, but let me show you why I still think this is worse.

Before, you can easily understand what is happening when composing lookups (as happens fairly often) by reading the line lineary left to right. But when we invert the logic it becomes harder to parse because you have to parse from the inside out.

For example (--- before, +++ after):
```diff
- cidx.const_(m).type_idx().type_(m)
+ m.type_(m.const_(*cidx).type_idx())
```

(Note also the extra deref)

And this is exasperated further when the arguments are not trivial:
```diff
- ft.arg_ty_idxs()[usize::from(*arg_idx)].type_(m)
+ m.type_(ft.arg_ty_idxs()[usize::from(*arg_idx)])
```

I think what we had before is a nicer API, but I won't argue further if you still disagree.

Reply:
The original API was equivalent to saying `0.lookup(&vec![])` rather than the more idiomatic `vec![].lookup(0)`, which is why I don't like the original.

In terms of the update, why not give each function a helper on `Module` so you have `func_type(...)` (etc.) rather than just `type_`? You end up with same terseness as the original, but without the unidiomatic API.

Reply:
Your way we shall do it ;)